src/main/kotlin/net/atos/zac/productaanvraag/ProductaanvraagService.kt

```diff
@@ -180,101 +180,177 @@ class ProductaanvraagService @Inject constructor(
     ) {
         var initiatorAdded = false
         productaanvraag.betrokkenen?.forEach {
-            when (it.rolOmschrijvingGeneriek) {
-                Betrokkene.RolOmschrijvingGeneriek.ADVISEUR -> {
-                    addBetrokkene(it, OmschrijvingGeneriekEnum.ADVISEUR, zaak)
-                }
+            initiatorAdded = if (it.roltypeOmschrijving != null) {
+                addBetrokkenenWithRole(it, initiatorAdded, zaak)
+            } else {
+                addBetrokkenenWithGenericRole(it, initiatorAdded, zaak)
+            }
+        }
+    }
 
-                Betrokkene.RolOmschrijvingGeneriek.BELANGHEBBENDE -> {
-                    addBetrokkene(it, OmschrijvingGeneriekEnum.BELANGHEBBENDE, zaak)
+    private fun addBetrokkenenWithRole(
+        betrokkene: Betrokkene,
+        initiatorAdded: Boolean,
+        zaak: Zaak
+    ): Boolean {
+        when (betrokkene.roltypeOmschrijving.lowercase()) {
+            Betrokkene.RolOmschrijvingGeneriek.INITIATOR.toString() -> {
+                if (initiatorAdded) {
+                    LOG.warning(
+                        "Multiple initiator betrokkenen found in productaanvraag for zaak '$zaak'. " +
+                            "Only the first one will be used."
+                    )
+                } else {
+                    addBetrokkene(betrokkene, OmschrijvingGeneriekEnum.INITIATOR.toString(), zaak)
                 }
+                return true
+            }
+            Betrokkene.RolOmschrijvingGeneriek.BEHANDELAAR.toString() -> {
+                LOG.warning(
+                    "Betrokkene with role '$betrokkene.roltypeOmschrijving' is not supported in the mapping from a " +
```

Comment:
This is not correct right? `'$betrokkene.roltypeOmschrijving'` should be `'${betrokkene.roltypeOmschrijving}'.

But in this case you already know it is BEHANDELAAR so why not use that? Much more readable.

Reply:
This one is for role description which is a text field. We can either compare it with equals(ignoreCase=true) or add additional constants.

Changed it to use `equals(ignoreCase=true)` now, but perhaps constants would be shorter and clearer

Reply:
Fixed